From 6262f49de177a79bc17f8d583aa5a7acaf48bf9c Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Fri, 28 Mar 2025 12:13:26 +0800
Subject: [PATCH] setup.py: Make the cmd available

The cmd will be None in OE environment as below.
 >>> import os
 >>> os.environ.get('CC', 'gcc')
'x86_64-wrs-linux-gcc  -m64 -march=nehalem -mtune=generic -mfpmath=sse -msse4.2 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/buildarea/tmp/work/corei7-64-wrs-linux/python3-m2crypto/0.44.0/recipe-sysroot'
 >>> import shutil
 >>> shutil.which(os.environ.get('CC', 'gcc'))
 >>> cmd = [shutil.which(os.environ.get('CC', 'gcc'))]
 >>> print(cmd)
[None]

So change the check logic to get the expected cmd.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 setup.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 792d7365b6ed..5b8a5d791793 100644
--- a/setup.py
+++ b/setup.py
@@ -208,8 +208,10 @@ class _M2CryptoBuildExt(build_ext.build_ext):
         if sys.platform != "win32":
             # generate src/SWIG/x509_v_flag.h to overcome weaknesses of swig
             # https://todo.sr.ht/~mcepl/m2crypto/298
-            with open("src/SWIG/x509_v_flag.h", "w", encoding="utf-8") as x509_v_h:
-                cmd = [shutil.which(os.environ.get("CC", "gcc"))]
+            with open(
+                "src/SWIG/x509_v_flag.h", "w", encoding="utf-8"
+            ) as x509_v_h:
+                cmd = os.environ.get('CC', 'gcc').split()
                 cflags = os.environ.get("CFLAGS")
                 if cflags is not None:
                     cmd += cflags.split()
-- 
2.34.1

